src/metabase/lib/metadata/columns.cljc

```diff
@@ -0,0 +1,526 @@
+(ns metabase.lib.metadata.columns
```

Comment:
This namespace is mostly ported over from `middleware.annotate`; I initially tried to update that namespace to use the new version here but it ended up being harder than I thought, be cause `annotate` uses the QP store rather than `:lib/metadata` attached to the query, and it is written with Schema and non-pipeline MBQL. So the changes are different enough that sharing it wasn't super easy.

I think this is effectively the new version of the namespace and at some point in the future when we update the QP to understand pipeline MBQL (pMBQL) and use Malli we can easily switch over to this -- the QP store can just be configured to be another metadata source